src/bin/cargo-flamegraph.rs

```diff
@@ -174,91 +176,86 @@ fn build(opt: &Opt) {
         cmd.arg("--no-default-features");
     }
 
+    cmd.arg("--message-format=json");
+
     if opt.verbose {
         println!("build command: {:?}", cmd);
     }
 
-    let mut child = cmd
-        .spawn()
-        .expect("failed to spawn cargo build command");
-
-    let exit_status = child.wait().expect(
-        "failed to wait for cargo build child to finish",
-    );
+    let Output {
+        status,
+        stdout,
+        stderr,
+    } = cmd
+        .output()
+        .expect("failed to execute cargo build command");
+
+    let messages = Message::parse_stream(&*stdout);
+    let artifacts: Vec<_> = messages
```

Comment:
Is there a reason to bind `messages` here? It looks like we can call `filter_map()` directly on the return value of `Message:parse_stream()`.

Reply:
No. I think I originally did this because the code didn't look good after `cargo fmt` because of the 60 characters max width in `.rustfmt.toml`. But it seems like I changed the code again so it looks just fine now.